app/models/redis_store.rb

```diff
@@ -16,19 +28,21 @@ def initialize(attributes = {}, persisted = false)
   end
 
   def self.find(redis_key = nil)
-    attributes = REDIS_STORE.hgetall(redis_key).with_indifferent_access
+    response = redis.get(redis_key)
+    return nil unless response
+    attributes = Oj.load(response)
     return nil if attributes.blank?
     object = new(attributes, true)
     if object.valid?
       object
     else
-      REDIS_STORE.del(redis_key)
+      redis.del(redis_key)
```

Comment:
redis_name_space or namespaced_store might be more clear, as redis might imply Redis.current which is not namespaced.